### ISSUE TITLE
[BUG] 7zip/unrar is non-free

### ISSUE THREAD

**Issue by linsui** [WirelessAlien/ZipXtract#28]
**Describe the bug**
See https://fedoraproject.org/wiki/Licensing:Unrar. Unrar is included in 7-Zip-JBinding-4Android. I'll disable affected versions. Could you please add a falvor without that lib or use a free alternative? https://github.com/zhanghai/libarchive-android also supports rar but it doesn't support encrypt 7z. For encrypt 7z you can build 7zip witout unrar.

**Comment by WirelessAlien** [comment]
> **Describe the bug** See https://fedoraproject.org/wiki/Licensing:Unrar. Unrar is included in 7-Zip-JBinding-4Android. I'll disable affected versions. Could you please add a falvor without that lib or use a free alternative? https://github.com/zhanghai/libarchive-android also supports rar but it doesn't support encrypt 7z. For encrypt 7z you can build 7zip witout unrar.

Yeah, I know but 7-zip> sevenzipjbind> 7-Zip-JBinding-4Android are same and under same licence and I found [7zip Licence](https://www.7-zip.org/license.txt) that - 

    unRAR license restriction
 
    The decompression engine for RAR archives was developed using source
    code of unRAR program.
    All copyrights to original unRAR code are owned by Alexander Roshal.

    The license for original unRAR code has the following restriction:

      The unRAR sources cannot be used to re-create the RAR compression algorithm,
      which is proprietary. Distribution of modified unRAR sources in separate form
      or as a part of other software is permitted, provided that it is clearly
      stated in the documentation and source comments that the code may
      not be used to develop a RAR (WinRAR) compatible archiver.

and in   -  https://fedoraproject.org/wiki/Licensing:Unrar

      The UnRAR sources may be used in any software to handle RAR
      archives without limitations free of charge, but cannot be used
      to re-create the RAR compression algorithm, which is proprietary.

We are not creating Rar archive, we are only extracting so I think it is allowed for free without restriction. Correct me if I am wrong.

**Comment by linsui** [comment]
It's allowed by the license but the license is not allowed in F-Droid. F-Droid requires the libs using FOSS license and unrar is not a FOSS license.

**Comment by WirelessAlien** [comment]
> It's allowed by the license but the license is not allowed in F-Droid. F-Droid requires the libs using FOSS license and unrar is not a FOSS license.

Ooh, Ok. Got it. I will add a flavour without using that library and [v4.0.2](https://github.com/WirelessAlien/ZipXtract/releases/tag/v4.0.2) is  the last version without sevenjipjbinding4android

**Comment by linsui** [comment]
Thanks!

**Comment by WirelessAlien** [comment]
> Thanks!

I am sorry, Just 1 question. Do I have to remove sevenzipjbinding4android completely, I mean I used sevenzipjbinding4android to create 7z also.

**Comment by linsui** [comment]
Yes, it's build with unrar included.

**Comment by WirelessAlien** [comment]
> Yes, it's build with unrar included.

Ok, thanks.

**Comment by WirelessAlien** [comment]
> Yes, it's build with unrar included.

I released a new version by removing sevenzipjbinding4android - https://github.com/WirelessAlien/ZipXtract/releases/tag/v4.3.2fd
target branch - fdroid :https://github.com/WirelessAlien/ZipXtract/tree/fdroid. Same package name, same signature, only app name changed from ZipXtract to ZipXtract FD
